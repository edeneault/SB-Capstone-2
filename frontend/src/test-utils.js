import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

// create a customRender that wraps the UI in a memory Router

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return <MemoryRouter initialEntries={initialRoutes}>{children}</MemoryRouter>;
};

const customRender = (ui, options) => {
  const initialRoutes =
    options && options.initialRoutes ? options.initialRoutes : ["/"];

  return render(ui, {
    wrapper: (args) =>
      MemoryRouterWithInitialRoutes({
        ...args,
        initialRoutes,
      }),
    ...options,
  });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
