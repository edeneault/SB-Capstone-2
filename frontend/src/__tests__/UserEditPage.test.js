import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { render } from "../test-utils";
import { initialState } from "../store-mock";
import UserEditPage from "../pages/UserEditPage.js";
import { createMemoryHistory } from "history";

const mockStore = configureMockStore([thunk]);

describe("test", () => {
  const history = createMemoryHistory();
  const path = `/admin/user/:id/edit`;

  const match = {
    isExact: false,
    path,
    url: path.replace(":id", "6125d04db949259bdec4de34"),
    params: { id: "6125d04db949259bdec4de34" },
  };

  test("renders UserEditPage without crashing", () => {
    const store = mockStore(initialState);

    const { getByText } = render(
      <Provider store={store}>
        <UserEditPage match={match} history={history} />
      </Provider>,
      { initialRoutes: ["/admin/user/6125d04db949259bdec4de34/edit"] },
    );
    // screen.debug();
    expect(getByText(/Edit User/i)).toBeInTheDocument();
  });

  test("matches snapshot", function () {
    const store = mockStore(initialState);

    const { asFragment } = render(
      <Provider store={store}>
        <UserEditPage match={match} history={history} loading={false} />
      </Provider>,
      { initialRoutes: [`/admin/user/6125d04db949259bdec4de34/edit`] },
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
