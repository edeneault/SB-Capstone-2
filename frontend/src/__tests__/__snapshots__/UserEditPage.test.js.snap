// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="btn btn-ligth my-3"
    href="/admin/usersList"
  >
    Go Back
  </a>
  <div
    class="container"
  >
    <div
      class="justify-content-md-center row"
    >
      <div
        class="col-md-6 col-12"
      >
        <h1>
          Edit User
        </h1>
        <form
          class=""
        >
          <div
            class="form-group"
          >
            <label
              class="form-label"
              for="name"
            >
              Name
            </label>
            <input
              class="form-control"
              id="name"
              placeholder="Enter name"
              type="name"
              value="AdminName"
            />
          </div>
          <div
            class="form-group"
          >
            <label
              class="form-label"
              for="email"
            >
              Email Address
            </label>
            <input
              class="form-control"
              id="email"
              placeholder="Enter email"
              type="email"
              value="admin@example.com"
            />
          </div>
          <div
            class="form-group"
          >
            <div
              class="form-check"
            >
              <input
                class="form-check-input"
                id="isAdmin"
                type="checkbox"
              />
              <label
                class="form-check-label"
                for="isAdmin"
                title=""
              >
                Is Admin
              </label>
            </div>
          </div>
          <button
            class="my-2 btn btn-primary"
            type="submit"
          >
            Update
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
